dns/paged_dns_store.go

```diff
@@ -0,0 +1,581 @@
+package main
+
+import (
+	"bytes"
+	"container/list"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/coreos/etcd/snap"
+
+	"github.com/miekg/dns"
+)
+
+// PageStoreMaxLen is max number of pages allowed in memory
+// Set this to 2 for easy debugging validation of LRU policy.
+const PageStoreMaxLen = 100
+
+// PageSize is the expected size inside of a page.
+// e.g. if PageSize = 1, there will be 2^63 pages
```

Comment:
2^64?

Reply:
It is 2^63. `PageSize` is probably a bit of a misnomer, since it is actually means how many lower bits we will ignore during page allocation (we use `64 - PageSize` most significant bits for a page number)